src/lug/lug.py

```diff
@@ -124,48 +149,65 @@ def patch_system_calls(func, user_docker_container_name, docker_shell_location):
         docker_shell_location=docker_shell_location,
     )
 
+    patched_functions = []
     find_and_replace_function(
         member=func.__globals__,
         unique_docstring="Execute a child program in a new process.",
-        replace_with=new_subprocess_popen
+        replace_with=new_subprocess_popen,
+        patched_functions=patched_functions,
     )
     if not hasattr(subprocess.Popen, "is_lug_function"):
         find_and_replace_function(
             member=func.__globals__,
             unique_docstring="Run command with arguments and return a CompletedProcess instance",
-            replace_with=new_subprocess_run
+            replace_with=new_subprocess_run,
+            patched_functions=patched_functions,
         )
     find_and_replace_function(
         member=func.__globals__,
         unique_docstring="Execute the command in a subshell",
-        replace_with=new_os_system
+        replace_with=new_os_system,
+        patched_functions=patched_functions,
     )
 
+    return patched_functions
 
-def unpatch_system_calls(func):
-    find_and_replace_function(
-        member=func.__globals__,
-        unique_docstring=None,
-        replace_with=None,
-        unpatch=True
-    )
+
+def unpatch_system_calls(patched_functions):
```

Comment:
This replaces the prior method of unpatching, which was calling `find_and_replace_function` with `unpatch=True`.

This keeps track of all patched functions, and then iterates through the tracked patched functions.